src/ui/pages/ConnectedSites/ConnectedSites.tsx

```diff
@@ -143,38 +148,120 @@ function ConnectedSitesList({
   );
 }
 
+export function ConnectedSitesSearch({
+  value,
+  onChange,
+}: {
+  value?: string;
+  onChange(value: string): void;
+}) {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const debouncedHandleChange = useDebouncedCallback(onChange, 300);
+
+  useLayoutEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.value = value || '';
+    }
+  }, [value]);
```

Comment:
What is this for?

Reply:
This is probably for Reset button. To make this input both uncontrolled and controlled

Reply:
The approach is directly inspired by the search on the history tab: `useLayoutEffect` ensures that `inputRef`'s value stays in sync with the `value` prop (particularly when it gets reset).

Now I've refactored `ConnectedSitesSearch` to be fully controlled and moved `inputRef` outside